packages/core-data/src/actions.js

```diff
@@ -677,6 +714,11 @@ export const saveEntityRecord =
 							),
 						};
 					}
+					// Unless there is no persisted record, set the status to
+					// publish.
+					if ( name === 'wp_template' && persistedRecord ) {
```

Comment:
This doesn't feel like the right place for this code. Should we have a similar behavior to pages/post with a publish flow :)

Reply:
Fixed in #72285.